src/renate/updaters/model_updater.py

```diff
@@ -125,15 +127,37 @@ def _load_best_checkpoint_and_save(self, trainer: Trainer, pl_module: LightningM
         # Reload best state.
         learner_state_path = Path(defaults.learner_state_file(self._output_state_folder))
         if learner_state_path.exists():
-            self._model.load_state_dict(torch.load(defaults.model_file(self.dirpath)))
-            pl_module.load_state_dict(self._model, torch.load(learner_state_path)["state_dict"])
+            loaded_state = trainer.strategy.load_checkpoint(learner_state_path)
+            pl_module.on_load_checkpoint(loaded_state)
+            # This loads the state dict only if its not Deepspeed.
+            # Deepspeed load the state_dict automatically when load_checkpoint is called.
+            trainer.strategy.load_model_state_dict(loaded_state)
+
         # Finalize model update.
         pl_module.on_model_update_end()
         # Save permanently.
         pl_module.save(self._output_state_folder)
         # Overwrite checkpoint.
         self._save_checkpoint(trainer, learner_state_path)
 
+    def teardown(self, trainer: Trainer, pl_module: LightningModule, stage: str) -> None:
```

Comment:
A doc string would be helpful here. I don't exactly follow what we do here. Is the goal to have a separate file that just contains the model weights?

Reply:
yes. added a detailed description